Src/VimMac/CommandKeyBinding.cs

```diff
@@ -0,0 +1,124 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+
+namespace Vim.VisualStudio
+{
+    /// <summary>
+    /// Id of an EnvDTE.Command object.  The Name portion of a Command is optional and doesn't
+    /// figure into the name.  The Guid and ID are unique to a command
+    /// </summary>
+    public readonly struct CommandId : IEquatable<CommandId>
+    {
+        public readonly Guid Group;
+
+        public readonly uint Id;
+
+        public CommandId(Guid group, uint id)
+        {
+            Group = group;
+            Id = id;
+        }
+
+        public bool Equals(CommandId other)
+        {
+            return Id == other.Id && Group == other.Group;
+        }
+
+        public override bool Equals(object obj)
+        {
+            if (obj is CommandId)

```

Comment:
Nit:
```csharp
return obj is CommandId cmd && Equals (cmd);
```

Reply:
This was just copied verbatim from a file that was probably written years ago.

I hope that at some point, once we figure out which code is common between VSMac and VS Windows, that we can come up with a plan for sharing the code instead of copying.

Reply:
> I hope that at some point, once we figure out which code is common between VSMac and VS Windows, that we can come up with a plan for sharing the code instead of copying.

Feel the same way. I think once we get the VSMac code merged and a bit of experience with developers using it, it will become fairly clear what can / should be shared with the Windows version. At that point we can start the work of defining that layer and moving VS Windows and VS Mac to depend on it.